stdlib/digest.mli

```diff
@@ -82,3 +84,88 @@ val from_hex : string -> t
     @raise Invalid_argument if the argument is not exactly 32 hexadecimal
    characters.
    @since 4.00 *)
+
+(** {1 Generic interface} *)
+
+module type S = sig
+
+  type t = string
+    (** The type of digests. *)
+
+  val hash_size : int
+    (** The length of digests, in bytes. *)
+
+  val compare : t -> t -> int
+    (** Compare two digests, , with the same specification as
+        {!Stdlib.compare}. *)
+
+  val equal : t -> t -> bool
+    (** Test two digests for equality. *)
+
+  val string : string -> t
+    (** Return the digest of the given string. *)
+
+  val bytes : bytes -> t
+    (** Return the digest of the given byte sequence. *)
+
+  val substring : string -> int -> int -> t
+    (** [substring s ofs len] returns the digest of the substring
+        of [s] starting at index [ofs] and containing [len] characters. *)
+
+  val subbytes : bytes -> int -> int -> t
+    (** [subbytes s ofs len] returns the digest of the subsequence
+        of [s] starting at index [ofs] and containing [len] bytes. *)
+
+  val channel : in_channel -> int -> t
+    (** Read characters from the channel and return their digest.
+        See {!Digest.channel} for the full specification. *)
+
+  val file : string -> t
+    (** Return the digest of the file whose name is given. *)
+
+  val output : out_channel -> t -> unit
+    (** Write a digest on the given output channel. *)
+
+  val input : in_channel -> t
+    (** Read a digest from the given input channel. *)
+
+  val to_hex : t -> string
+    (** Return the printable hexadecimal representation of the given digest.
+        @raise Invalid_argument if the length of the argument
+        is not [hash_size], *)
+
+  val from_hex : string -> t
```

Comment:
Maybe we could tweak that to `of_hex` for consistency and simply have the toplevel `Digest` module have both `of_hex` and `from_hex`.

Reply:
Done in [002a780](https://github.com/ocaml/ocaml/pull/12307/commits/002a7805a0ab6b7c1d3e6850f245b7a5984f2b31).